README_NEW.md

```diff
@@ -0,0 +1,349 @@
+```
+
+## Common information
+
+### MLflow source code
+
+To integrate with MLflow, you need to include the source code. The [mlflow/mlflow](https://github.com/mlflow/mlflow/) repository contains proto files that define the tracking API. It also includes Python tests that we use to verify our Go implementation produces identical behaviour.
+
+We use a `.mlflow.ref` file to specify the exact location from which to pull our sources. The format should be `remote#reference`, where `remote` is a git remote and `reference` is a branch, tag, or commit SHA.
+
+If the `.mlflow.ref` file is modified and becomes out of sync with the current source files, the mage target will automatically detect this. To manually force a sync, you can run `mage repo:update`.
+
+### Protos
+
+To ensure we stay compatible with the Python implementation, we aim to generate as much as possible based on the `.proto` files.
+
+By running
+
+```bash
+mage generate
+```
+
+Go code will be generated. Use the protos files from `.mlflow.repo` repository.
+
+This includes the generation of:
+
+- Structs for each endpoint. ([pkg/protos](./protos/service.pb.go))
+- Go interfaces for each service. ([pkg/contract/service/*.g.go](./contract/service/tracking.g.go))
+- [fiber](https://gofiber.io/) routes for each endpoint. ([pkg/server/routes/*.g.go](./server/routes/tracking.g.go))
+
+If there is any change in the proto files, this should ripple into the Go code.
+
+### Request validation
+
+We use [Go validator](https://github.com/go-playground/validator) to validate all incoming request structs.
+As the proto files don't specify any validation rules, we map them manually in [pkg/cmd/generate/validations.go](./cmd/generate/validations.go).
+
+Once the mapping has been done, validation will be invoked automatically in the generated fiber code.
+
+When the need arises, we can write custom validation function in [pkg/validation/validation.go](./validation/validation.go).
+
+### Data access
+
+Initially, we want to focus on supporting Postgres SQL. We chose [Gorm](https://gorm.io/) as ORM to interact with the database.
+
+We do not generate any Go code based on the database schema. Gorm has generation capabilities but they didn't fit our needs. The plan would be to eventually assert the current code stil matches the database schema via an intergration test.
+
+All the models use pointers for their fields. We do this for performance reasons and to distinguish between zero values and null values.
+
+### Linting
+
+We have enabled various linters from [golangci-lint](https://golangci-lint.run/), you can run these via:
+
+```bash
+pre-commit run golangci-lint --all-files
+```
+
+Sometimes `golangci-lint` can complain about unrelated files, run `golangci-lint cache clean` to clear the cache.
+
+## Usage
+
+### Go server
```

Comment:
For developers it would be interesting to mention `mage dev`, that is how I always start the thing.